### PR TITLE
Enable eclipse compatible restart for wells with gas lift optimization

### PR DESCRIPTION
**Pull request by jalvestad**
This PR implements writing of Eclipse compatible restart file data for the keywords: LITOPT and WLIFTOPT. 

The changes enables Eclipse restart when these keywords are used in the model.
The changes involves writing information to the following arrays:
INTEHEAD
DOUBHEAD
IWEL
SWEL

Unit tests remains to be added. 

Flow restart from flow-simulation, runs, however one of the wells in a test-dataset shuts at the very beginning of the restart simulation. This is probably caused by a well modelling problem, and need to be invesigated to identify the problem.

### CODE REVIEW

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/output/eclipse/DoubHEAD.cpp [hunk 1]
```diff
@@ -154,10 +154,10 @@ enum Index : std::vector<double>::size_type {
     grpar_d =  VI::doubhead::GRpar_d,
     grpar_e =  VI::doubhead::GRpar_e,
     grpar_f =  VI::doubhead::GRpar_f,
-    dh_093  =  93,
-    dh_094  =  94,
+    LOmini  =  VI::doubhead::LOminInt,
```
Comment (joakim-hove) [inline src/opm/output/eclipse/DoubHEAD.cpp:157]: Good with named constants!

**Comment by jalvestad** [comment]
Unit tests have been added for the relevant restart data. The pr is considered ready for final review and merge from my point of view.

**Comment by joakim-hove** [comment]
I have not looked in detail - but is it really necessary to add a 50.000 lines testcase?

**Comment by joakim-hove** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Two questions: 
1. I can not judge whether the test failures are to be expected or not - although we seem to have updates in models which do not use gas lift optimization? So good if @jalvestad will comment on the test failures. 

2. As mentioned I thought the 50.000 lines of testcase code was excessive; but if it is really needed then we merge it.

When these points are resolved I am ready to do the required data update and merge.

**Comment by bska** [comment]
For what it's worth, I've run this patch locally and verified that the only change to the restart files is `INTEHEAD[59]` changing from zero to two.  I'll run a build check to make sure that the CI system agrees.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
> I am ready to do the required data update and merge.

Please hold for OPM/opm-simulators#2949 and OPM/opm-tests#455 to avoid conflicting updates.  We must also decide which of this and #2164 to merge first because both will require updating the reference solutions and they will generate conflicting updates.

**Comment by bska** [comment]
@joakim-hove : Do you have any reservation with respect to merging this?  Otherwise, I'd like to generate new reference solutions to bring this into master.

**Comment by bska** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
>Do you have any reservation with respect to merging this?

No - merge away!

**Comment by bska** [comment]
> merge away!

Much appreciated!  I'll generate new reference solutions now.

**Comment by bska** [comment]
jenkins build this update_data please

**Comment by bska** [comment]
jenkins build this opm-tests=463 please

**Pull request merged**